packages/lodestar/src/network/gossip/gossip.ts

```diff
@@ -33,6 +33,7 @@ import {IBeaconChain} from "../../chain";
 import {computeEpochAtSlot, computeForkDigest} from "@chainsafe/lodestar-beacon-state-transition";
 import {GossipEncoding} from "./encoding";
 import {toHexString} from "@chainsafe/ssz";
+import {Libp2p} from "libp2p-gossipsub/src/interfaces";
```

Comment:
gossipsub has its own libp2p type, libp2p-interface uses `any`. Also gossipsub uses peer-id 0.14.0 and lodestar uses 0.14.1 (drived by other dependencies) and it causes type conflict with this minor version change.
I heard that libp2p has a plan to generate its type from jsdoc like libp2p-interface, can we wait for it? Temporarily made it as ts-ignore to avoid the confusion and version conflict issue.

Reply:
> I heard that libp2p has a plan to generate its type from jsdoc like libp2p-interface, can we wait for it?

Nope, it's last one that's gonna receive this update and it's probably gonna take a long time. Why aren't we using libp2p type from ts-libp2p? Since libp2p-interfaces has any type it's gonna work

Reply:
gossipsub-js uses its own libp2p type. @wemeetagain can we use libp2p-ts libp2p type in gossipsub-js?